### PR TITLE
Migrating to use any container (not only mouf) + badges + PSR-2

### PR DESCRIPTION
**Pull request by moufmouf**

### CODE REVIEW

File: src/Mouf/Html/Renderer/AutoChainRenderer.php [hunk 1]
```diff
@@ -1,271 +1,279 @@
...
+                    $isCachable = false;
+                }
+                if ($result == ChainableRendererInterface::CAN_RENDER_OBJECT || $result == ChainableRendererInterface::CAN_RENDER_CLASS) {
+                    $foundRenderer = $renderer;
+                    break 2;
+                }
+            }
+        } while (false);
+
+        if ($isCachable && $foundRenderer) {
```
Comment (cbalda) [inline src/Mouf/Html/Renderer/AutoChainRenderer.php:128]: Did you mean "Catchable" ?

File: src/Mouf/Html/Renderer/AutoChainRenderer.php [hunk 2]
```diff
@@ -1,271 +1,279 @@
...
+            // TODO: suboptimal
```
Comment (cbalda) [inline src/Mouf/Html/Renderer/AutoChainRenderer.php]: Is a "TODO" comment clean in your code ?

File: src/Mouf/Html/Renderer/AutoChainRenderer.php (hunk 1 above)
Comment (moufmouf) [inline src/Mouf/Html/Renderer/AutoChainRenderer.php:128]: Nope, I meant "cachable" like "is it possible to put it in the cache".

File: src/Mouf/Html/Renderer/FileBasedRenderer.php [hunk 3]
```diff
@@ -1,261 +1,282 @@
...
+        if ($container === null) {
+            $container = MoufManager::getMoufManager();
+        }
+
+        $this->twig->addExtension(new MoufTwigExtension($container));
+        $this->twig->addExtension(new \Twig_Extension_Debug());
+    }
+
+    /**
+     * (non-PHPdoc)
```
Comment (cbalda) [inline src/Mouf/Html/Renderer/FileBasedRenderer.php]: Comments ?

File: src/Mouf/Html/Renderer/FileBasedRenderer.php [hunk 4]
```diff
@@ -1,261 +1,282 @@
...
+
+        if ($fileName) {
+            return ChainableRendererInterface::CAN_RENDER_CLASS;
+        } else {
+            return ChainableRendererInterface::CANNOT_RENDER;
+        }
+    }
+
+    /**
+     * (non-PHPdoc)
```
Comment (cbalda) [inline src/Mouf/Html/Renderer/FileBasedRenderer.php]: Comments

File: src/Mouf/Html/Renderer/FileBasedRenderer.php [hunk 5]
```diff
@@ -1,261 +1,282 @@
...
+
+        $this->canRender($object, $context);
+
+        $this->debugFalse = true;
+
+        return $this->debugStr;
+    }
+
+    /**
+     * (non-PHPdoc)
```
Comment (cbalda) [inline src/Mouf/Html/Renderer/FileBasedRenderer.php]: Comments

File: src/Mouf/Html/Renderer/AutoChainRenderer.php (hunk 1 above)
Comment (cbalda) [inline src/Mouf/Html/Renderer/AutoChainRenderer.php:128]: Ok

**Pull request merged**